Sources/MapboxMaps/Annotations/ViewAnnotation.swift

```diff
@@ -0,0 +1,240 @@
+    /// Annotation `options` must include Geometry where we want to bind our view annotation.
+    ///
+    /// Width and height could be specified explicitly but better idea will be not specifying them
+    /// as they will be calculated automatically based on view layout.
+    ///
+    /// - Parameters:
+    ///   - view: `UIView` to be wrapped in an `AnnotationView` and placed on the map
+    ///   - options: view annotation options
+    ///
+    /// - Throws: `ViewAnnotationManagerError.insertionFailure` if options did not include geometry or contains incomplatible values.
+    public func addAnnotationView(withContent view: UIView, options: ViewAnnotationOptions) throws -> AnnotationView {
+        guard options.geometry != nil else {
+            throw ViewAnnotationManagerError.insertionFailure(reason: "Geometry field is missing")
+        }
+        var creationOptions = options
+        if creationOptions.width == nil {
+            creationOptions.width = view.bounds.size.width
+        }
+        if creationOptions.height == nil {
+            creationOptions.height = view.bounds.size.height
+        }
+        let annotatonView = AnnotationView(view: view, annotationManager: self)
+        try mapboxMap.addViewAnnotation(withId: annotatonView.id, options: options)
+        annotationViewsById[annotatonView.id] = annotatonView
+        annotationIdByViews[annotatonView] = annotatonView.id
+        containerView.addSubview(annotatonView)
+        return annotatonView
+    }
+
+    /// Remove given `AnnotationView` from the map if it was present.
+    ///
+    /// - Throws:
+    ///   - `ViewAnnotationManagerError.annotationNotFound`: the supplioed view was not found
+    ///   - `ViewAnnotationManagerError.removalFailure`: errors during the removal of the view
+    public func remove(_ annotatonView: AnnotationView) throws {
+        guard let id = annotationIdByViews[annotatonView], let annotatonView = annotationViewsById[id] else {
+            throw ViewAnnotationManagerError.annotationNotFound
+        }
+        try mapboxMap.removeViewAnnotation(withId: id)
+        annotatonView.removeFromSuperview()
+        annotationViewsById.removeValue(forKey: id)
+        annotationIdByViews.removeValue(forKey: annotatonView)
+    }
+
+    /// Update given `AnnotationView` with `ViewAnnotationOptions`.
+    /// Important thing to keep in mind that only properties present in `options` will be updated,
+    /// all other will remain the same as specified before.
+    ///
+    /// - Throws:
+    ///   - `ViewAnnotationManagerError.annotationNotFound`: the supplioed view was not found
+    ///   - `ViewAnnotationManagerError.updateFailure`: errors during the up date of the view (eg. incorrect parameters)
+    public func update(_ annotatonView: AnnotationView, options: ViewAnnotationOptions) throws {
+        guard let id = annotationIdByViews[annotatonView] else {
+            throw ViewAnnotationManagerError.annotationNotFound
+        }
```

Comment:
Is this error handling redundant? Could we just let gl-native throw if the id isn't found?